### PR TITLE
OCM-1388 | fix: add retry count to throttle retry message

### PR DESCRIPTION
**Pull request by davidleerh**

### CODE REVIEW

File: pkg/aws/client.go [hunk 1]
```diff
@@ -1256,7 +1256,8 @@ func (r CustomRetryer) ShouldRetry(req *request.Request) bool {
 	}
 	logger := logging.NewLogger()
 	if strings.Contains(req.Error.Error(), "Throttling") {
-		logger.Warn("Throttling Rate limit exceeded. Retrying the request again")
+		logger.Warn(fmt.Sprintf("Throttling Rate limit exceeded. Retrying Attempt: '%v' of '%v'",
+			req.RetryCount+1, r.MaxRetries()))
```
Comment (davidleerh) [inline pkg/aws/client.go:1260]: After manual test, it seems the retry count starts at index 0.

**Review by willkutler — APPROVED** [review]

**Review by robpblake — APPROVED** [review]
/lgtm

**Pull request merged**